addon/components/help-widget/chat.js

```diff
@@ -65,9 +65,18 @@ export default class HelpWidgetChatComponent extends Component {
   }
 
   @action
-  onKeydown(_text, e) {
+  updateTentativeMessage({ target }) {
+    this.tentativeMessage = target.value;
```

Comment:
I'm no ember expert, but couldn't this be achieved with `@tracked`?

Reply:
`this.tenativeMessage` _is_ `@tracked`. It's the value of the input (or in this case `textarea`, I think) that isn't inherently tracked by the system. So we have to bind to the `input` event of the `<input/>` or `<textarea/>` and update the tracked variable ourselves.

Ember _does_ have builtin helpers that do this already: `<Input/>` and `<Textarea/>`. HOWEVER! We need to do respond to key events on those elements (like pressing "Enter"). Normally, this works fine: you just pass `@enter={{this.someAction}}` and go on your merry way. However, when inserted into Members, this is apparently more complicated. 😵 

There are two ways Ember has historically handled events. One is to have a sort of "global" listener which dispatches events to where they need to go in the system. The Ember `<body/>` tag or whatever root div has all the listeners and then when an event bubbles up, it consults its own registry on how to dispatch them. This is how things were historically done in Ember and how, I think, events still work for Ember's builtin `<Input/>` and `<Textarea/>` components.

The other way is to have listeners on the elements themselves and respond to everything locally. This is how the `{{on}}` modifier works -- it's just syntactic sugar over `element.addEventListener(...)`. This is the newer, more modern and lightweight way of listening for events.

To make a long story slightly shorter, _something_ in Members interferes with the global event handling method, whereas the local event handlers work just fine. I wasn't quite able to track down exactly _where_ in Members we were intercepting some sort of `keydown`, `keyup`, or `keypress` (did you know "keypress" is deprecated? I didn't, but apparently it is!) event that prevents the global event dispatch from functioning, but I suspect this will be an issue with builtin Ember input components until we find it and deal with it.

Anyhow, the benefit of using the builtin input components is that their `@value` is two-way bound -- any updates to the input are automatically populated into the variable you passed in; in our case, `this.tentativeMessage` would be updated automatically! But the downside is that they're components under-the-hood, and you can't assign modifiers to components. This means we had no way of using local event dispatch for listening for "Enter" being pressed and could only rely on the global dispatch which wasn't working. So, even though it meant having to "roll our own" two-way binding using a method like this, it ultimately was the better way to go, since it meant we could respond to Enter in Members.

Maybe more of an explanation than you'd bargained for, but I'd meant to bring this up anyways, as it's going to be an issue with Ember in Members until we figure out where the interference is coming from.

Reply:
Great explanation! Thanks!